### PR TITLE
feat(vscode): add command to select server config from history

### PR DESCRIPTION
**Pull request by wsxiaoys**
fix https://github.com/TabbyML/tabby/issues/2962

cc @liangfung

### CODE REVIEW

File: clients/vscode/src/CommandPalette.ts [hunk 1]
```diff
@@ -119,7 +119,8 @@ export default class CommandPalette {
       }
       item.command = () => this.issues.showHelpMessage();
     } else if (agentStatus === "ready") {
-      item.label = "Ready";
+      item.label = `Ready`;
```
Comment (wsxiaoys) [inline clients/vscode/src/CommandPalette.ts:122]:
```suggestion
      item.label = "Ready";
```

File: clients/vscode/package.json [hunk 2]
```diff
@@ -133,6 +133,11 @@
         "command": "tabby.chat.edit.discard",
         "title": "Discard Changes",
         "category": "Tabby"
+      },
+      {
+        "command": "tabby.server.selectPastServerConfig",
+        "title": "Select Server Config from History",
```
Comment (wsxiaoys) [inline clients/vscode/package.json:139]:
```suggestion
        "title": "Select Server Endpoint from History",
```

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 3]
```diff
@@ -136,6 +136,7 @@ export class TabbyAgent extends EventEmitter implements Agent {
     }
 
     if (!this.api || isServerConnectionChanged) {
+      this.changeStatus("notInitialized");
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts:139]: @icycodes this is required to didChangeStatus will be triggered.

Before this change, if user switching config A to config B, and each config are legit, there won't be another `ready` status being triggered due to event deduping check.

Comment (icycodes) [reply clients/tabby-agent/src/TabbyAgent.ts:139]: LGTM

**Pull request merged**